### DISCUSSION TITLE
after mig to vite: fetcher.load triggers full reload, when server returns promise without json()

### DISCUSSION THREAD

**Discussion by akomm** [remix-run/remix#9549] [Q&A]
- vite 5.2.12
- remix 2.9.2
- react 18.3.1 / 18.3-canary
- no single fetch or other unstable features enabled. 
- using remix-flat-routes

The redirect is not triggered by server, all status 200 and no location headers. It must be the fetcher that triggers the site reload. It only happens after migrating to vite and ruling out all issues (no errors displayed in console / server output).

Without vite there is no full page reload.

The problem: I can't reproduce this with an empty repository, including flat-routes. And sadly I can't share the large codebase :(. 

Because of the dependence on plain promise returned vs. await & json(), I can't but think that it might be related to the API changes in preparation for RSC / single fetch that are expected to support promises over the wire.

**Comment by asgeo1** [comment]
@akomm I'm having what I think is the same issue. If I use `remix dev` instead of `remix vite:dev` the issue goes away.

Did you ever work this out?

**Reply by akomm** [reply]
no

**Comment by asgeo1** [comment]
@akomm I think I've worked out my issue.

First, I turned on "preserve log" in Chrome Dev Tools, and then I noticed there was indeed an error in the console.

<img width="1074" alt="image" src="https://github.com/user-attachments/assets/cc87cf5e-fd4d-4e12-8114-e3efd98a685d">

Because it immediately refreshes the page, it was hard to notice this.

The error message itself doesn't make any sense to me. When I examined the route itself, I worked out that the issue was caused by this:

`routes/api.account.exists.ts`
```typescript
export const someFunction = async () => {
  ...
}

export const action = async ({ request }: ActionFunctionArgs) => {
  ...
}
```

If I remove the `export` from in front of `someFunction`, then the issue would go away.  I never specifically meant to export that function from the route, it was just an oversight.

But doing that with `vite` causes the error above and page refresh. Where as on the `classic` compiler it works fine.

Maybe this was your issue too?

**Reply by akomm** [reply]
I don't have extra exports in my routes. But its been a while ago. I don't recall how I worked around the issue.
